test/router-options.test.js

```diff
@@ -58,3 +63,25 @@ test('Should honor maxParamLength option', t => {
     t.strictEqual(res.statusCode, 404)
   })
 })
+
+test('Should honor frameworkErrors option', t => {
+  t.plan(1)
+  const fastify = Fastify({
+    frameworkErrors: function (error, req, res) {
+      if (error instanceof FST_ERR_BAD_URL) {
+        t.ok(true)
+      } else {
+        t.fail()
+      }
+    }
+  })
+
+  fastify.get('/test/:id', (req, reply) => {
+    reply.send({ hello: 'world' })
+  })
+
+  fastify.inject({
+    method: 'GET',
+    url: '/test/%world'
+  })
```

Comment:
we should check the result here

Reply:
I'm sorry I don't understand what do you mean

Reply:
I mean

```
  fastify.inject({
    method: 'GET',
    url: '/test/%world'
  }, (err, res) => {
t.error(err)
t.deepEquals(res.json(), ....)
} )
```